app/src/main/java/com/bogdan/codeforceswatcher/features/problems/redux/requests/ProblemsRequests.kt

```diff
@@ -50,6 +66,16 @@ class ProblemsRequests {
             return problems
         }
 
+        private fun bindProblemsToContests(problems: List<Problem>) {
+            val contests = store.state.contests.contests
+            for (problem in problems) {
+                contests.find { it.id == problem.contestId }?.let { contest ->
```

Comment:
I'm not sure but it can N*M complexity. Can be greatly improved by using map of <ContestId, Contest>.

Reply:
Hey, optimizations)